app/src/main/java/emu/skyline/input/onscreen/OnScreenEditActivity.kt

```diff
@@ -54,35 +61,103 @@ class OnScreenEditActivity : AppCompatActivity() {
+    }
+
+    private fun createBackgroundColorPicker() : ColorPicker {
+        return ColorPicker(this@OnScreenEditActivity).apply {
+            setTitle(this@OnScreenEditActivity.getString(R.string.osc_background_color))
+            setRoundColorButton(true)
+            setColors(*colors.toIntArray())
+            setDefaultColorButton(binding.onScreenControllerView.getBGColor())
+            positiveButton.visibility = View.GONE
+            negativeButton.visibility = View.GONE
+            setOnChooseColorListener(object : ColorPicker.OnChooseColorListener {
+                override fun onChooseColor(position : Int, color : Int) {
+                    binding.onScreenControllerView.setBGColor(colors[position])
+                }
+
+                override fun onCancel() {/*Nothing to do*/
+                }
+            })
+            show()
+            dismissDialog()
+        }
+    }
+
+    private fun createTextColorPicker(backgroundColorPicker : ColorPicker) : ColorPicker {
+        return ColorPicker(this@OnScreenEditActivity).apply {
+            setTitle(this@OnScreenEditActivity.getString(R.string.osc_text_color))
+            setRoundColorButton(true)
+            setColors(*colors.toIntArray())
+            setDefaultColorButton(binding.onScreenControllerView.getTextColor())
+            setOnChooseColorListener(object : ColorPicker.OnChooseColorListener {
+                override fun onChooseColor(position : Int, color : Int) {
+                    binding.onScreenControllerView.setTextColor(colors[position])
+                    backgroundColorPicker.positiveButton.performClick()
+                }
+
+                override fun onCancel() {/*Nothing to do*/
+                }
+            })
+            show()
+        }
+    }
+
+    private fun mergeColorPickers(mainColorPicker : ColorPicker, secondColorPicker : ColorPicker) {
+        with(mainColorPicker) {
+            dialogViewLayout.apply {
+                layoutParams = FrameLayout.LayoutParams(FrameLayout.LayoutParams.WRAP_CONTENT, FrameLayout.LayoutParams.WRAP_CONTENT).apply { gravity = Gravity.BOTTOM }
+                if (viewTreeObserver.isAlive) {
+                    viewTreeObserver.addOnGlobalLayoutListener(object : OnGlobalLayoutListener {
+                        override fun onGlobalLayout() {
+                            viewTreeObserver.removeOnGlobalLayoutListener(this)
+                            getmDialog()!!.window!!.setLayout((width * 1.2).roundToInt(), (height * 1.8).roundToInt())
+                        }
+                    })
+                }
+            }
+            (secondColorPicker.dialogViewLayout.parent as ViewGroup).removeView(secondColorPicker.dialogViewLayout)
+            getmDialog()?.addContentView(secondColorPicker.dialogViewLayout, LinearLayout.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT))
+        }
+    }
```

Comment:
You definitely don't want to do this to have both color pickers in the same dialog window. I'm not sure about the library you're using, but you should be able to easily create a new XML layout with both color sections in there, then hook up the code to load up that XML. If you can't do that, you'll need to draw them in code, but there's no need to show both, then dismiss one, then merge them manually like you are doing here.

Reply:
I had to fork the project and create my own color picker with double selector, it was not as easy as i expected but that's done and it works